### PR TITLE
Implemented favouriteThis Use Case

### PR DESCRIPTION
**Pull request by ryanzyi**
This pull request contains the commits for the favouriteThis use case. 

The implementation should follow SOLID and Clean Architecture best practices. Please let me know if there is any discrepancy between the new code and best practices. 

Tests are still being written, and I plan to finish them by this Wednesday.

UI changes were implemented, including a button and a dialog message pop up. They will be updated in the next couple of days.

### CODE REVIEW

File: src/view/CookThisOrReRollView.java [hunk 1]
```diff
@@ -127,6 +168,25 @@ private void updateView(CookThisOrReRollState cookThisOrReRollState) {
...
+                    JOptionPane.INFORMATION_MESSAGE);
+        } else {
+//            favouriteButton.setIcon(favouriteIconOff);
+            System.out.println("Off");
+            JOptionPane.showMessageDialog(null,
+                    "This recipe was unfavourited",
+                    "Recipe unfavourited",
+                    JOptionPane.INFORMATION_MESSAGE);
+        }
+        //titleImagePanel, otherComponentsPanel
```
Comment (HC123abc) [inline src/view/CookThisOrReRollView.java:189]: -Good you have 2 favourited or unfavourited

File: src/use_case/favouriteThis/favouriteThisInteractor.java [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+                fTInputData.getServings(),fTInputData.getSummary(), fTInputData.getIngredients(),
+                fTInputData.getInstructions(), fTInputData.getImage(), fTInputData.getId());
+
+        if (isFavourite) {
+            favouriteRecipes.removeOneRecipe(recipe);
+            fTOutputData = new favouriteThisOutputData(false);
+
+        } else {
+            favouriteRecipes.setOneFavouriteRecipe(recipe);
+            fTOutputData = new favouriteThisOutputData(true);
```
Comment (HC123abc) [inline src/use_case/favouriteThis/favouriteThisInteractor.java:44]: Good that you have a output data and does not return a entity. Follows Clean Architecture

**Review by HC123abc — COMMENTED** [review]
Overall Very Good.
- Code follow style convention and Naming convention. 
- Follow Solid, you create a new use case for favorited, also you have all the interface for dependency inversion
- Follow Clean architecture
- Very very good

**Comment by HC123abc** [comment]
There is a merge conflict but in Tutorial we can fix it

File: src/view/CookThisOrReRollView.java (hunk 1 above)
Comment (ryanzyi) [reply src/view/CookThisOrReRollView.java:189]: I will clean up the code in next few commits and make a new pull request with testing 👍

**Comment by ryanzyi** [comment]
Fixed the merge conflicts. They were minor, except for the individually made changes Harley and I did to the CookThisOrReRollView constructor. I've kept Harley's implementation and modify it to include my code.

**Review by TakumaInnami — COMMENTED** [review]
Thank you Ryan! Now the conflict is fully solved and we can merge the pull.

**Comment by stwig** [comment]
Like everyone said, naming convention might need a quick look, and also yea try to make it a bit cleaner (you can make helpers for view, etc). Overall very good though! 🐡

**Pull request merged**